grep/flags.rb

```diff
@@ -0,0 +1,57 @@
+require 'io/console'
+
+class Flags
+  def initialize(flags, files)
+    @result = []
+    @flags = flags
+    @files = files
+  end
+
+  def read_files(files)
+    src = {}
+    files.each { |name| src[name] = File.read(name) }
+    src
+  end
+
+  def grep(str)
+    pattern = create_pattern(str)
+    @flags.include?('-v') ? v_flag(pattern) : f(pattern)
+    @flags.include?('-l') && @result.length > 1 ? @result.uniq : @result
+  end
+
+  private
+
+  def f(pattern)
+    @files.each do |file|
+      IO.readlines(file).each_with_index do |string, index|
+        @result.push(check_flags(file, string, index)) if string.match(pattern)
+      end
+    end
+  end
+
+  def v_flag(pattern)
+    @files.each do |file|
+      IO.readlines(file).each_with_index do |string, index|
+        @result.push(check_flags(file, string, index)) unless string.match(pattern)
+      end
+    end
+  end
+
```

Comment:
methods are the totally the same, except if/unless.
common logic can be retrieved to one single method.